patches/components-permissions-android-permission_dialog_delegate.cc.patch

```diff
@@ -1,12 +1,18 @@
 diff --git a/components/permissions/android/permission_dialog_delegate.cc b/components/permissions/android/permission_dialog_delegate.cc
-index 19905a0a9f24afdfe4bd6f78d03be2a633d18cba..200439fe161f652de764bfc0d66d5cf4c816690a 100644
+index 19905a0a9f24afdfe4bd6f78d03be2a633d18cba..c5a7414edd9fce805d5d3acd32b1fa17e2f4fc1b 100644
 --- a/components/permissions/android/permission_dialog_delegate.cc
 +++ b/components/permissions/android/permission_dialog_delegate.cc
-@@ -67,6 +67,7 @@ void PermissionDialogDelegate::CreateJavaDelegate(
+@@ -67,11 +67,13 @@ void PermissionDialogDelegate::CreateJavaDelegate(
  
  void PermissionDialogDelegate::Accept(JNIEnv* env,
                                        const JavaParamRef<jobject>& obj) {
 +  BRAVE_PERMISSION_DIALOG_DELEGATE_ACCEPT
    permission_prompt_->Accept();
  }
  
+ void PermissionDialogDelegate::Cancel(JNIEnv* env,
+                                       const JavaParamRef<jobject>& obj) {
++  BRAVE_PERMISSION_DIALOG_DELEGATE_CANCEL
```

Comment:
I think with `Cancel` you can redefine the method in `chromium_src` for .h and .cc. It wasn't possible for `Accept` since it also calls `Accept` internally, but `Cancel` calls `Deny`, so maybe we can avoid this part of the patch?

Reply:
```
#define Cancel                                                        \
  Cancel_ChromiumImpl(JNIEnv* env, const JavaParamRef<jobject>& obj); \
  void Cancel(JNIEnv* env, const JavaParamRef<jobject>& obj);
```
this doesn't work because `Cancel` is widely used everywhere and I see some weird errors:
```
In file included from ../../mojo/public/cpp/bindings/lib/interface_ptr_state.h:31:
In file included from ../../mojo/public/cpp/bindings/lib/multiplex_router.h:29:
In file included from ../../mojo/public/cpp/bindings/connector.h:24:
In file included from ../../mojo/public/cpp/system/handle_signal_tracker.h:12:
../../mojo/public/cpp/system/simple_watcher.h:143:8: error: no template named 'JavaParamRef'; did you mean 'base::android::JavaParamRef'?
  void Cancel();
```
not sure how it can be done the other way, because it's a JNI method and JNI-generated code expects exactly `PermissionDialogDelegate::Cancel` method. I don't see an option to make it virtual with a define because it's also widely used.

are there some other options I don't see?